tremor-script/src/srs.rs

```diff
@@ -111,6 +239,15 @@ impl Debug for Query {
 }
 
 impl Query {
+    /// Creates a new Query with a pre-existing query
+    #[must_use]
+    pub fn new_from_ast(structured: ast::Query) -> Self {
```

Comment:
this is not sound we need to capture the the Arc/pin that `ast::Query` was created with or we risk referencing freed memory

Reply:
Sound-ness is one of the things we need to go through - and document carefully. I suspected
as much but wasn't certain that we'd need to capture the underlying pinned reference.

Reply:
I write a long explanation in the issue https://github.com/tremor-rs/tremor-runtime/pull/1104 which should probably have ended up in the file. but in short the guarantees of the SRS **only** hold true and offer a sound use of transmuting when for every clone, move other extraction of data the corresponding vac+arc+pin is also cloned. That is what guarantees that the referenced data isn't freed until the referencing data is freed. That's where this two functions fall short, they'll need to include the vecarcpin (we need a word for this :joy: )

Reply:
Great - this is the context I was missing and my brain still harks back to rental/transmute here as we kept that interface which I think I tripped up on!